src/lib/store-forms.module.ts

```diff
@@ -22,23 +26,19 @@ export const defaultStoreFormsConfig: StoreFormsConfig = {
   ],
   exports: [
     BindingDirective
-  ],
-  providers: [{
-    provide: STORE_FORMS_CONFIG,
-    useValue: defaultStoreFormsConfig
-  }, {
-    provide: BindingService,
-    useClass: BindingService,
-    deps: [STORE_FORMS_CONFIG, Store]
-  }]
+  ]
 })
 export class StoreFormsModule {
   static forRoot(config?: StoreFormsConfig): ModuleWithProviders {
     return {
       ngModule: StoreFormsModule,
       providers: [{
+        provide: _STORE_FORMS_CONFIG,
+        useValue: config
+      }, {
         provide: STORE_FORMS_CONFIG,
```

Comment:
This seams weird... Isn't there any other more elegant solution for this problem?

Reply:
Not that I know of, for example it's not possible to return a function expression in a decorator. Also checked other libraries and it looks like that they're using the same approach.